src/server/scripts/EasternKingdoms/zone_eversong_woods.cpp

```diff
@@ -0,0 +1,216 @@
+                    break;
+                case 4:
+                    me->LoadEquipment(urand(1, 2));
+                    me->HandleEmoteCommand(EMOTE_ONESHOT_EAT_NO_SHEATHE);
+                    _events.ScheduleEvent(EVENT_REMOVE_EQUIPMENT_PATHER, 4s);
+                    break;
+                case 5:
+                    me->SetUInt32Value(UNIT_NPC_EMOTESTATE, EMOTE_STATE_DANCE);
+                    _events.ScheduleEvent(EVENT_STOP_DANCING_PATHER, 6s);
+                    break;
+                }
+                break;
+            }
+            case EVENT_REMOVE_EQUIPMENT_PATHER:
+                me->LoadEquipment(0, true);
+                _events.ScheduleEvent(EVENT_PATH, 8s);
+                break;
+            case EVENT_STOP_DANCING_PATHER:
+                me->SetUInt32Value(UNIT_NPC_EMOTESTATE, EMOTE_ONESHOT_NONE);
+                _events.ScheduleEvent(EVENT_PATH, 5s);
+                break;
+                break;
+            }
+        }
+
+        if (!UpdateVictim())
+            return;
+
+        DoMeleeAttackIfReady();
+    }
+
+private:
+    EventMap _events;
+    uint32   _path;
+};
+
+enum Partygoer
+{
+    EVENT_RANDOM_ACTION     = 1,
+    EVENT_REMOVE_EQUIPMENT,
+    EVENT_STOP_DANCING,
+    EVENT_THROW_FIREWORKS,
+    EVENT_RESET_FACING,
+
+    GO_FIREWORKS_LAUNCHER   = 180771,
+
+    SPELL_SMALL_YELLOW_ROCKET = 26295
+};
+
+struct npc_partygoer : public ScriptedAI
+{
+    npc_partygoer(Creature* creature) : ScriptedAI(creature) { }
+
+    void JustAppeared() override
+    {
+        _facing = me->GetOrientation();
+        _events.ScheduleEvent(EVENT_RANDOM_ACTION, 1s, 20s);
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        if (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+            case EVENT_RANDOM_ACTION:
+            {
+                int8 _action = urand(1, 6);
+
+                switch (_action)
+                {
+                case 1:
+                    me->HandleEmoteCommand(EMOTE_ONESHOT_TALK);
+                    _events.ScheduleEvent(EVENT_RANDOM_ACTION, 13s, 20s);
+                    break;
+                case 2:
+                    me->HandleEmoteCommand(EMOTE_ONESHOT_EXCLAMATION);
+                    _events.ScheduleEvent(EVENT_RANDOM_ACTION, 13s, 20s);
+                    break;
+                case 3:
+                    me->HandleEmoteCommand(EMOTE_ONESHOT_LAUGH);
+                    _events.ScheduleEvent(EVENT_RANDOM_ACTION, 13s, 20s);
+                    break;
+                case 4:
+                    me->LoadEquipment(urand(1, 2));
+                    me->HandleEmoteCommand(EMOTE_ONESHOT_EAT_NO_SHEATHE);
+                    _events.ScheduleEvent(EVENT_REMOVE_EQUIPMENT, 4s);
+                    break;
+                case 5:
+                    me->SetUInt32Value(UNIT_NPC_EMOTESTATE, EMOTE_STATE_DANCE);
```

Comment:
Do not set UNIT_NPC_EMOTESTATE manually by directly using SetUInt32Value, use `SetEmoteState(EMOTE_STATE_DANCE)`

Reply:
Will fix after I eat.

Reply:
all 4 fixed